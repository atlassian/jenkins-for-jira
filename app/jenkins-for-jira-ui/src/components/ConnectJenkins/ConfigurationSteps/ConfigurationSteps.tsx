import React from 'react';
import { Status } from '@atlaskit/progress-tracker/dist/types/types';
// TODO: Find out why the Atlassian and Forge packages break ESLint
// eslint-disable-next-line
import { ProgressTracker, Stages } from '@atlaskit/progress-tracker';
import { useHistory } from 'react-router';
import { progressTrackerContainer } from './ConfigurationSteps.styles';

type AppProps = {
	currentStage: string;
};

const mapCreateStatus = (currentStage: unknown): Status => {
	switch (currentStage) {
		case 'create':
			return 'current';
		case 'connect':
			return 'visited';
		default:
			return 'unvisited';
	}
};

// This is not reusable for other configuration steps.
// Choosing to apply YAGNI for now.
const ConfigurationSteps = ({ currentStage }: AppProps) => {
	const history = useHistory();

	const stages: Stages = [
		{
			id: 'install',
			label: 'Install plugin',
			percentageComplete:
				currentStage === 'install'
					? 0
					: 100,
			status:
				currentStage === 'install'
					? 'current'
					: 'visited',
			href: '#',
			onClick: () => { history.push('/install'); }
		},
		{
			id: 'create',
			label: 'Create server',
			percentageComplete:
				currentStage === 'create' || currentStage === 'install'
					? 0
					: 100,
			status: mapCreateStatus(currentStage),
			href: '#'
		},
		{
			id: 'connect',
			label: 'Connect server',
			percentageComplete: 0,
			status:
				currentStage === 'connect'
					? 'current'
					: 'unvisited',
			href: '#'
		}
	];

	return (
		<div className={progressTrackerContainer}>
			<ProgressTracker items={stages} />
		</div>
	);
};

export {
	ConfigurationSteps
};
