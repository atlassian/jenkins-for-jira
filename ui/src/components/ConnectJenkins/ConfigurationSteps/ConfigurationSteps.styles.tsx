import { css } from '@emotion/css';

export const progressTrackerContainer = css`
	width: 184px;
`;
