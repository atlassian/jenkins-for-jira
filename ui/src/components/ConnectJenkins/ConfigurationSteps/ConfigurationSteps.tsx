import React from 'react';
// TODO: Find out why the Atlassian and Forge packages break ESLint
// eslint-disable-next-line
import { ProgressTracker, Stages } from '@atlaskit/progress-tracker';
import { useHistory } from 'react-router';
import { progressTrackerContainer } from './ConfigurationSteps.styles';

type AppProps = {
	currentStage: string;
};

// This is not reusable for other configuration steps.
// Choosing to apply YAGNI for now.
const ConfigurationSteps = ({ currentStage }: AppProps) => {
	const history = useHistory();

	const stages: Stages = [
		{
			id: 'install',
			label: 'Install plugin',
			percentageComplete:
				currentStage === 'install'
					? 0
					: 100,
			status:
				currentStage === 'install'
					? 'current'
					: 'visited',
			href: '#',
			onClick: () => { history.push('/install'); }
		},
		{
			id: 'connect',
			label: 'Connect your app',
			percentageComplete: 0,
			status:
				currentStage === 'connect'
					? 'current'
					: 'unvisited',
			href: '#'
		}
	];

	return (
		<div className={progressTrackerContainer}>
			<ProgressTracker items={stages} />
		</div>
	);
};

export {
	ConfigurationSteps
};
