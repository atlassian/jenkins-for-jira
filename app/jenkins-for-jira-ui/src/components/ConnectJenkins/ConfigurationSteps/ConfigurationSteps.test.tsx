import { render, screen } from '@testing-library/react';
import { ConfigurationSteps } from './ConfigurationSteps';

describe('ConfigurationSteps Page Suite', () => {
	const { getByText } = screen;

	it('Should render three stages', () => {
		render(<ConfigurationSteps currentStage="install" />);
		expect(getByText('Install plugin')).toBeInTheDocument();
		expect(getByText('Create server')).toBeInTheDocument();
		expect(getByText('Connect server')).toBeInTheDocument();
	});

	it('Should render correctly with install as currentStage', () => {
		const container = render(<ConfigurationSteps currentStage="install" />);
		const InstallJenkinsStep = getByText('Install plugin');
		const CreateJenkinsStep = getByText('Create server');
		const ConnectJenkinsStep = getByText('Connect server');

		expect(InstallJenkinsStep).toHaveAttribute('style', 'color: rgb(0, 101, 255); font-weight: 600;');
		expect(CreateJenkinsStep).toHaveAttribute('style', 'color: rgb(94, 108, 132); font-weight: 400;');
		expect(ConnectJenkinsStep).toHaveAttribute('style', 'color: rgb(94, 108, 132); font-weight: 400;');
		expect(container).toMatchSnapshot();
	});

	it('Should render correctly with create as currentStage', () => {
		const container = render(<ConfigurationSteps currentStage="create" />);
		const InstallJenkinsStep = getByText('Install plugin');
		const CreateJenkinsStep = getByText('Create server');
		const ConnectJenkinsStep = getByText('Connect server');

		expect(InstallJenkinsStep).toBeInTheDocument();
		expect(CreateJenkinsStep).toHaveAttribute('style', 'color: rgb(0, 101, 255); font-weight: 600;');
		expect(ConnectJenkinsStep).toHaveAttribute('style', 'color: rgb(94, 108, 132); font-weight: 400;');
		expect(container).toMatchSnapshot();
	});

	it('Should render correctly with connect as currentStage', () => {
		const container = render(<ConfigurationSteps currentStage="connect" />);
		const InstallJenkinsStep = getByText('Install plugin');
		const CreateJenkinsStep = getByText('Create server');
		const ConnectJenkinsStep = getByText('Connect server');

		expect(InstallJenkinsStep).toBeInTheDocument();
		expect(CreateJenkinsStep).toBeInTheDocument();
		expect(ConnectJenkinsStep).toHaveAttribute('style', 'color: rgb(0, 101, 255); font-weight: 600;');
		expect(container).toMatchSnapshot();
	});
});
