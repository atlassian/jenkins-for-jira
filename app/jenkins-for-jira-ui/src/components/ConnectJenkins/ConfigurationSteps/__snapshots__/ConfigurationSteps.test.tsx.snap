// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigurationSteps Page Suite Should render correctly with connect as currentStage 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-17crxen"
      >
        <ul
          aria-label="Progress"
          class="css-t7ahcc"
          style="grid-template-columns: repeat(3, 1fr); --ds--pt--sp: 16px; max-width: 480px;"
        >
          <li
            aria-current="false"
            class="css-ln603f"
            style="--ds--pt--ts: 50ms; --ds--pt--td: 0ms; --ds--pt--te: linear; --ds--pt--mc: #0065FF; --ds--pt--bg: #0065FF;"
          >
            <div
              class="css-1kxonj9"
            >
              <div
                class="css-gilxpy-ProgressMarker fade-appear fade-appear-active"
              />
              <div
                class="css-15hmvtc-ProgressBar fade-appear fade-appear-active"
              />
              <div
                class="css-9d8jpc fade-appear fade-appear-active"
                style="color: rgb(23, 43, 77); font-weight: 600;"
              >
                <a
                  class="css-diynhl-Link"
                  href="#"
                >
                  Install plugin
                </a>
              </div>
            </div>
          </li>
          <li
            aria-current="false"
            class="css-ln603f"
            style="--ds--pt--ts: 50ms; --ds--pt--td: 50ms; --ds--pt--te: linear; --ds--pt--mc: #0065FF; --ds--pt--bg: #0065FF;"
          >
            <div
              class="css-1kxonj9"
            >
              <div
                class="css-gilxpy-ProgressMarker fade-appear fade-appear-active"
              />
              <div
                class="css-15hmvtc-ProgressBar fade-appear fade-appear-active"
              />
              <div
                class="css-9d8jpc fade-appear fade-appear-active"
                style="color: rgb(23, 43, 77); font-weight: 600;"
              >
                <a
                  class="css-diynhl-Link"
                  href="#"
                >
                  Create server
                </a>
              </div>
            </div>
          </li>
          <li
            aria-current="step"
            class="css-ln603f"
            style="--ds--pt--ts: 50ms; --ds--pt--td: 0ms; --ds--pt--te: linear; --ds--pt--mc: #0065FF; --ds--pt--bg: #0065FF;"
          >
            <div
              class="css-1kxonj9"
            >
              <div
                class="css-gilxpy-ProgressMarker fade-appear fade-appear-active"
              />
              <div
                class="css-15hmvtc-ProgressBar fade-appear fade-appear-active"
              />
              <div
                class="css-9d8jpc fade-appear fade-appear-active"
                style="color: rgb(0, 101, 255); font-weight: 600;"
              >
                Connect server
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-17crxen"
    >
      <ul
        aria-label="Progress"
        class="css-t7ahcc"
        style="grid-template-columns: repeat(3, 1fr); --ds--pt--sp: 16px; max-width: 480px;"
      >
        <li
          aria-current="false"
          class="css-ln603f"
          style="--ds--pt--ts: 50ms; --ds--pt--td: 0ms; --ds--pt--te: linear; --ds--pt--mc: #0065FF; --ds--pt--bg: #0065FF;"
        >
          <div
            class="css-1kxonj9"
          >
            <div
              class="css-gilxpy-ProgressMarker fade-appear fade-appear-active"
            />
            <div
              class="css-15hmvtc-ProgressBar fade-appear fade-appear-active"
            />
            <div
              class="css-9d8jpc fade-appear fade-appear-active"
              style="color: rgb(23, 43, 77); font-weight: 600;"
            >
              <a
                class="css-diynhl-Link"
                href="#"
              >
                Install plugin
              </a>
            </div>
          </div>
        </li>
        <li
          aria-current="false"
          class="css-ln603f"
          style="--ds--pt--ts: 50ms; --ds--pt--td: 50ms; --ds--pt--te: linear; --ds--pt--mc: #0065FF; --ds--pt--bg: #0065FF;"
        >
          <div
            class="css-1kxonj9"
          >
            <div
              class="css-gilxpy-ProgressMarker fade-appear fade-appear-active"
            />
            <div
              class="css-15hmvtc-ProgressBar fade-appear fade-appear-active"
            />
            <div
              class="css-9d8jpc fade-appear fade-appear-active"
              style="color: rgb(23, 43, 77); font-weight: 600;"
            >
              <a
                class="css-diynhl-Link"
                href="#"
              >
                Create server
              </a>
            </div>
          </div>
        </li>
        <li
          aria-current="step"
          class="css-ln603f"
          style="--ds--pt--ts: 50ms; --ds--pt--td: 0ms; --ds--pt--te: linear; --ds--pt--mc: #0065FF; --ds--pt--bg: #0065FF;"
        >
          <div
            class="css-1kxonj9"
          >
            <div
              class="css-gilxpy-ProgressMarker fade-appear fade-appear-active"
            />
            <div
              class="css-15hmvtc-ProgressBar fade-appear fade-appear-active"
            />
            <div
              class="css-9d8jpc fade-appear fade-appear-active"
              style="color: rgb(0, 101, 255); font-weight: 600;"
            >
              Connect server
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ConfigurationSteps Page Suite Should render correctly with create as currentStage 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-17crxen"
      >
        <ul
          aria-label="Progress"
          class="css-t7ahcc"
          style="grid-template-columns: repeat(3, 1fr); --ds--pt--sp: 16px; max-width: 480px;"
        >
          <li
            aria-current="false"
            class="css-ln603f"
            style="--ds--pt--ts: 300ms; --ds--pt--td: 0ms; --ds--pt--te: cubic-bezier(0.15,1,0.3,1); --ds--pt--mc: #0065FF; --ds--pt--bg: #0065FF;"
          >
            <div
              class="css-1kxonj9"
            >
              <div
                class="css-gilxpy-ProgressMarker fade-appear fade-appear-active"
              />
              <div
                class="css-15hmvtc-ProgressBar fade-appear fade-appear-active"
              />
              <div
                class="css-9d8jpc fade-appear fade-appear-active"
                style="color: rgb(23, 43, 77); font-weight: 600;"
              >
                <a
                  class="css-diynhl-Link"
                  href="#"
                >
                  Install plugin
                </a>
              </div>
            </div>
          </li>
          <li
            aria-current="step"
            class="css-ln603f"
            style="--ds--pt--ts: 300ms; --ds--pt--td: 0ms; --ds--pt--te: cubic-bezier(0.15,1,0.3,1); --ds--pt--mc: #0065FF; --ds--pt--bg: #0065FF;"
          >
            <div
              class="css-1kxonj9"
            >
              <div
                class="css-gilxpy-ProgressMarker fade-appear fade-appear-active"
              />
              <div
                class="css-15hmvtc-ProgressBar fade-appear fade-appear-active"
              />
              <div
                class="css-9d8jpc fade-appear fade-appear-active"
                style="color: rgb(0, 101, 255); font-weight: 600;"
              >
                Create server
              </div>
            </div>
          </li>
          <li
            aria-current="false"
            class="css-ln603f"
            style="--ds--pt--ts: 300ms; --ds--pt--td: 0ms; --ds--pt--te: cubic-bezier(0.15,1,0.3,1); --ds--pt--mc: #A5ADBA; --ds--pt--bg: #A5ADBA;"
          >
            <div
              class="css-1kxonj9"
            >
              <div
                class="css-gilxpy-ProgressMarker fade-appear fade-appear-active"
              />
              <div
                class="css-15hmvtc-ProgressBar fade-appear fade-appear-active"
              />
              <div
                class="css-9d8jpc fade-appear fade-appear-active"
                style="color: rgb(94, 108, 132); font-weight: 400;"
              >
                Connect server
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-17crxen"
    >
      <ul
        aria-label="Progress"
        class="css-t7ahcc"
        style="grid-template-columns: repeat(3, 1fr); --ds--pt--sp: 16px; max-width: 480px;"
      >
        <li
          aria-current="false"
          class="css-ln603f"
          style="--ds--pt--ts: 300ms; --ds--pt--td: 0ms; --ds--pt--te: cubic-bezier(0.15,1,0.3,1); --ds--pt--mc: #0065FF; --ds--pt--bg: #0065FF;"
        >
          <div
            class="css-1kxonj9"
          >
            <div
              class="css-gilxpy-ProgressMarker fade-appear fade-appear-active"
            />
            <div
              class="css-15hmvtc-ProgressBar fade-appear fade-appear-active"
            />
            <div
              class="css-9d8jpc fade-appear fade-appear-active"
              style="color: rgb(23, 43, 77); font-weight: 600;"
            >
              <a
                class="css-diynhl-Link"
                href="#"
              >
                Install plugin
              </a>
            </div>
          </div>
        </li>
        <li
          aria-current="step"
          class="css-ln603f"
          style="--ds--pt--ts: 300ms; --ds--pt--td: 0ms; --ds--pt--te: cubic-bezier(0.15,1,0.3,1); --ds--pt--mc: #0065FF; --ds--pt--bg: #0065FF;"
        >
          <div
            class="css-1kxonj9"
          >
            <div
              class="css-gilxpy-ProgressMarker fade-appear fade-appear-active"
            />
            <div
              class="css-15hmvtc-ProgressBar fade-appear fade-appear-active"
            />
            <div
              class="css-9d8jpc fade-appear fade-appear-active"
              style="color: rgb(0, 101, 255); font-weight: 600;"
            >
              Create server
            </div>
          </div>
        </li>
        <li
          aria-current="false"
          class="css-ln603f"
          style="--ds--pt--ts: 300ms; --ds--pt--td: 0ms; --ds--pt--te: cubic-bezier(0.15,1,0.3,1); --ds--pt--mc: #A5ADBA; --ds--pt--bg: #A5ADBA;"
        >
          <div
            class="css-1kxonj9"
          >
            <div
              class="css-gilxpy-ProgressMarker fade-appear fade-appear-active"
            />
            <div
              class="css-15hmvtc-ProgressBar fade-appear fade-appear-active"
            />
            <div
              class="css-9d8jpc fade-appear fade-appear-active"
              style="color: rgb(94, 108, 132); font-weight: 400;"
            >
              Connect server
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ConfigurationSteps Page Suite Should render correctly with install as currentStage 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="css-17crxen"
      >
        <ul
          aria-label="Progress"
          class="css-t7ahcc"
          style="grid-template-columns: repeat(3, 1fr); --ds--pt--sp: 16px; max-width: 480px;"
        >
          <li
            aria-current="step"
            class="css-ln603f"
            style="--ds--pt--ts: 300ms; --ds--pt--td: 0ms; --ds--pt--te: cubic-bezier(0.15,1,0.3,1); --ds--pt--mc: #0065FF; --ds--pt--bg: #0065FF;"
          >
            <div
              class="css-1kxonj9"
            >
              <div
                class="css-gilxpy-ProgressMarker fade-appear fade-appear-active"
              />
              <div
                class="css-15hmvtc-ProgressBar fade-appear fade-appear-active"
              />
              <div
                class="css-9d8jpc fade-appear fade-appear-active"
                style="color: rgb(0, 101, 255); font-weight: 600;"
              >
                Install plugin
              </div>
            </div>
          </li>
          <li
            aria-current="false"
            class="css-ln603f"
            style="--ds--pt--ts: 300ms; --ds--pt--td: 0ms; --ds--pt--te: cubic-bezier(0.15,1,0.3,1); --ds--pt--mc: #A5ADBA; --ds--pt--bg: #A5ADBA;"
          >
            <div
              class="css-1kxonj9"
            >
              <div
                class="css-gilxpy-ProgressMarker fade-appear fade-appear-active"
              />
              <div
                class="css-15hmvtc-ProgressBar fade-appear fade-appear-active"
              />
              <div
                class="css-9d8jpc fade-appear fade-appear-active"
                style="color: rgb(94, 108, 132); font-weight: 400;"
              >
                Create server
              </div>
            </div>
          </li>
          <li
            aria-current="false"
            class="css-ln603f"
            style="--ds--pt--ts: 300ms; --ds--pt--td: 0ms; --ds--pt--te: cubic-bezier(0.15,1,0.3,1); --ds--pt--mc: #A5ADBA; --ds--pt--bg: #A5ADBA;"
          >
            <div
              class="css-1kxonj9"
            >
              <div
                class="css-gilxpy-ProgressMarker fade-appear fade-appear-active"
              />
              <div
                class="css-15hmvtc-ProgressBar fade-appear fade-appear-active"
              />
              <div
                class="css-9d8jpc fade-appear fade-appear-active"
                style="color: rgb(94, 108, 132); font-weight: 400;"
              >
                Connect server
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="css-17crxen"
    >
      <ul
        aria-label="Progress"
        class="css-t7ahcc"
        style="grid-template-columns: repeat(3, 1fr); --ds--pt--sp: 16px; max-width: 480px;"
      >
        <li
          aria-current="step"
          class="css-ln603f"
          style="--ds--pt--ts: 300ms; --ds--pt--td: 0ms; --ds--pt--te: cubic-bezier(0.15,1,0.3,1); --ds--pt--mc: #0065FF; --ds--pt--bg: #0065FF;"
        >
          <div
            class="css-1kxonj9"
          >
            <div
              class="css-gilxpy-ProgressMarker fade-appear fade-appear-active"
            />
            <div
              class="css-15hmvtc-ProgressBar fade-appear fade-appear-active"
            />
            <div
              class="css-9d8jpc fade-appear fade-appear-active"
              style="color: rgb(0, 101, 255); font-weight: 600;"
            >
              Install plugin
            </div>
          </div>
        </li>
        <li
          aria-current="false"
          class="css-ln603f"
          style="--ds--pt--ts: 300ms; --ds--pt--td: 0ms; --ds--pt--te: cubic-bezier(0.15,1,0.3,1); --ds--pt--mc: #A5ADBA; --ds--pt--bg: #A5ADBA;"
        >
          <div
            class="css-1kxonj9"
          >
            <div
              class="css-gilxpy-ProgressMarker fade-appear fade-appear-active"
            />
            <div
              class="css-15hmvtc-ProgressBar fade-appear fade-appear-active"
            />
            <div
              class="css-9d8jpc fade-appear fade-appear-active"
              style="color: rgb(94, 108, 132); font-weight: 400;"
            >
              Create server
            </div>
          </div>
        </li>
        <li
          aria-current="false"
          class="css-ln603f"
          style="--ds--pt--ts: 300ms; --ds--pt--td: 0ms; --ds--pt--te: cubic-bezier(0.15,1,0.3,1); --ds--pt--mc: #A5ADBA; --ds--pt--bg: #A5ADBA;"
        >
          <div
            class="css-1kxonj9"
          >
            <div
              class="css-gilxpy-ProgressMarker fade-appear fade-appear-active"
            />
            <div
              class="css-15hmvtc-ProgressBar fade-appear fade-appear-active"
            />
            <div
              class="css-9d8jpc fade-appear fade-appear-active"
              style="color: rgb(94, 108, 132); font-weight: 400;"
            >
              Connect server
            </div>
          </div>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
